### PR TITLE
New Station Activity: Shipbreaking

### PR DESCRIPTION
**Pull request by cowbot92**
# Document the changes in your pull request

This will allow crew members to call in, and break down ships for resources, research points, goodies, and possibly more. For now this will be a box(tm) specific feature, but if enough people like it I'll map it to other maps. This will replace the useless podbay

Lore:  These scrapped ships being available to be scrapped is because NT invested a bunch of money into dredging up these ships from various places to fix up and repair them codenamed “shiptest” and then the guy in charge of the project got laid off and they cut the program so now there’s the new program "Shipbreaker"

# Why is this good for the game?

Gives people without specific set jobs something to do over the shift, or parts of it. Not focused on sec/antag gear but still rewarding.

# Testing

https://github.com/yogstation13/Yogstation/assets/75333826/26d5b260-35b7-4ed9-9854-4628f974bd69

Area where you'll go to begin ship breaking
image

image

<details closed>
<summary>Ship Designs</summary>
<br>
Syndicate Prison Transport

image

NT Med Transport

image

Old Robotics Transport

image

</details>

# Wiki Documentation

Wiki please reach out to me

# Changelog

:cl:  Cowbot92 & Chubbygummibear
rscadd: Adds new station side activity: Shipbreaking
wip: This is work in progress, feel free to comment ideas & ship "themes" or make your own for me to add
mapping: Removes the box podbay
experimental: This is very experimental
/:cl:

### CODE REVIEW

**Comment by Beeper-Skeeper** [comment]
OH
I GET IT NOW
THIS SHIT LOOKS FIRE

**Comment by AMyriad** [comment]
I really really like this idea, and yet I have a really big gripe with making this something that'd spawn stationside and give the public access to EVA equipment, or easily-stealable-by-hacking EVA equipment. I really believe it'd be best to make this a cargo job, rather than relegating it to the makeshift shipbreaking brigade of assistants. Assistants and engineers, because I can't think of anyone else who'd go out and disassemble ships.

I don't want to be a negative nelly here cause love the idea and I respect the work you're doin, but this could be better implemented than a podbay replacement of all things.

**Comment by wonderinghost** [comment]
> I really really like this idea, and yet I have a really big gripe with making this something that'd spawn stationside and give the public access to EVA equipment, or easily-stealable-by-hacking EVA equipment. I really believe it'd be best to make this a cargo job, rather than relegating it to the makeshift shipbreaking brigade of assistants. Assistants and engineers, because I can't think of anyone else who'd go out and disassemble ships.
> 
> I don't want to be a negative nelly here cause love the idea and I respect the work you're doin, but this could be better implemented than a podbay replacement of all things.

Could have this be like the miners conscription kit. Or possibly make up a bunch of jobs (unlisted) with their own set of accesses within departments. 

Also you could just move pod area over to where custom shuttle resources are

**Comment by wonderinghost** [comment]
Oh yeah you were asking for ship ideas.

The galaga ship could be one

**Comment by drderp3635** [comment]
I would add floors to the docking area like the dock on the mining outpost, so assistants can scrap without needing spacesuits.

File: yogstation/code/modules/shipbreaker/shipbreaker.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/datum/map_template/shipbreaker
+	should_place_on_top = FALSE
```
Comment (ToasterBiome) [inline yogstation/code/modules/shipbreaker/shipbreaker.dm]: these should be placed on top so shipbreaking can be spawned on stuff like asteroid turfs..

File: yogstation/code/game/objects/effects/landmarks.dm [hunk 2]
```diff
@@ -246,4 +246,10 @@ GLOBAL_LIST_EMPTY(chosen_station_templates)
 	GLOB.infiltrator_objective_items += loc
 	return INITIALIZE_HINT_QDEL 
 
+
+/obj/effect/landmark/shipbreak
+	name = "shipbreak spawnpoint"
+	icon_state = "Tourist"
```
Comment (ToasterBiome) [inline yogstation/code/game/objects/effects/landmarks.dm]: make this the generic one

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
All of the ships should use turf/template_noop instead of space too

File: yogstation/code/modules/shipbreaker/shipbreaker.dm (hunk 1 above)
Comment (cowbot92) [reply yogstation/code/modules/shipbreaker/shipbreaker.dm]: ok

File: yogstation/code/game/objects/effects/landmarks.dm (hunk 2 above)
Comment (cowbot92) [reply yogstation/code/game/objects/effects/landmarks.dm]: this landmark doesnt get used, removed

File: yogstation/code/modules/shipbreaker/shipbreaker.dm (hunk 1 above)
Comment (cowbot92) [reply yogstation/code/modules/shipbreaker/shipbreaker.dm]: updated

File: yogstation/code/game/objects/effects/landmarks.dm (hunk 2 above)
Comment (cowbot92) [reply yogstation/code/game/objects/effects/landmarks.dm]: removed unused landmark

**Comment by cowbot92** [comment]
> All of the ships should use turf/template_noop instead of space too

Updated

**Comment by wonderinghost** [comment]
this reminds me, got to do a bounty or work on updating the recycler so it checks the recipe, so everything doesnt produce just glass and iron

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
map review:

fix the recycler in disposals
make sure any windows have plating under them, not reinforced floor

**Comment by cowbot92** [comment]
> map review:
> 
> fix the recycler in disposals make sure any windows have plating under them, not reinforced floor

fixed and ship windows adjusted

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**